lib/rex/proto/http/packet.rb

```diff
@@ -324,6 +324,9 @@ def parse_header
           # So if we haven't seen either a Content-Length or a
           # Transfer-Encoding header, there shouldn't be a message body.
           self.body_bytes_left = 0
+        elsif (self.headers['Connection']&.downcase == 'upgrade' && self.headers['Upgrade']&.downcase == 'websocket')
```

Comment:
:+1:

I also double checked that the potential of having extra metadata in the Upgrade header as well, similar to `Content-type: application/json; charset=utf-8`, but confirmed that it wasn't a possibility. Potentially there could be surrounding whitespace, which I believe should be dropped according to the spec - but I believe the headers already drop whitespace as well :+1: